File: 6.Python/sentimental-cash/cash.py
from cs50 import get_float


def main():
    change = get_change()
    coins = calc_coins(change * 100)        # multiply by 100 to avoid floating point errors
    print(f"{coins:.0f}")                   # :.0f is how many decimal places to print


def get_change():
    while True:
        n = get_float("Change: ")
        if n > 0:
            return n


def calc_coins(change):
    coins = [25, 10, 5, 1]
    n = 0

    for coin in coins:
        n += change // coin                 # '//' is floor division. Rounds down to the neearest whole number.
        change %= coin                      # % returns the remainder after dividing (uses floor division).

    return n


main()

File: 1.C/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calc_coins(int change);

int main(void)
{
    int change;
    do
    {
        change = get_int("Change: ");
    }
    while (change < 0);

    int coins = calc_coins(change);
    printf("%i\n", coins);
}

int calc_coins(int change)
{
    int coin_value[4] = {25, 10, 5, 1};
    int n = 0;

    for (int i = 0; i < 4; i++)
    {
        n += change / coin_value[i];
        change -= coin_value[i] * (change / coin_value[i]);
    }
    return n;
}
